### PR TITLE
Scraping Dictionarycom for the Word of the Day

### PR DESCRIPTION
**Pull request by swetha3456**
## Description

Scrapes Dictionary.com for the word of the day and its definition.

## Checklist

> Before submitting this pull request, kindly verify that the ensuing checkpoints have been reached.

- [x] Have you adhered to the repository's defined coding convention rules?
- [x] Have you updated the 'documentation.md' file with the method/function documentation?
- [x] Have you sent a message along with the result or response?
- [x] Have you used the try-catch technique?
- [x] Has the method/class been added to the documentation (md file)?

## Screenshots

> Uploading a screenshot illustrating the approach you developed for validation is required.

image

## Additional Notes/Comments

> Any additional remarks or suggestions concerning this pull request are welcome.

---

I certify that I have carried out the relevant checks and provided the requisite screenshot for validation by submitting this pull request.
I appreciate your contribution.

### CODE REVIEW

**Review by nikhil25803 — CHANGES_REQUESTED** [review]
Requested some changes @swetha3456.

**Comment by swetha3456** [comment]
@nikhil25803, I've made the changes you requested. Please review my PR and suggest further edits if any. Thank you!

File: src/scrape_up/dictionary/dictionary.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
+from bs4 import BeautifulSoup
+from urllib.request import urlopen
+
+class Dictionary:
+    """
+    Create an instance of the `Dictionary` class.
+    ```python
+    scraper = IMDB()
```
Comment (nikhil25803) [inline src/scrape_up/dictionary/dictionary.py:8]: `Dictionary` class, not `IMDB` class.

File: src/scrape_up/dictionary/dictionary.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    ```
+    | Methods        | Details                                      |
+    | -------------- | -------------------------------------------- |
+    | `.get_word_of_the_day()` | Returns word of the day from Dictionary.com. |
+    | `.word_of_the_day_definition()` | Returns the definition of the word of the day. |
+    """
+    def get_word_of_the_day_url(self):
+        with urlopen("https://www.dictionary.com/") as response:
```
Comment (nikhil25803) [inline src/scrape_up/dictionary/dictionary.py:16]: Each method also do have their docstring, add them as well with syntax and sample response.

File: src/scrape_up/dictionary/dictionary.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        Returns a string containing the definition of the word of the day.
+        """
+        with urlopen(self.get_word_of_the_day_url()) as response:
+            soup = BeautifulSoup(response, 'html.parser')
+
+            for para in soup('p'):
+                if para.string and para.string[0] not in "EG":
+                    return para.string
+
+        return None
```
Comment (nikhil25803) [inline src/scrape_up/dictionary/dictionary.py:44]: Add `exception-handling` in all the methods created to deal with any exception occurred.

**Review by nikhil25803 — CHANGES_REQUESTED** [review]
Few more changes ...

**Comment by swetha3456** [comment]
Hi @nikhil25803, thanks for pointing out those errors! I've fixed those too. If anything else is left, please let me know.

**Review by nikhil25803 — APPROVED** [review]

**Comment by nikhil25803** [comment]
Great work @swetha3456 !!
Do not forget to
+ Share your contribution on socials and tag Clueless 🫂
+ Give this repository a star ✨
+ Follow Clueless Community on GitHub ✅

**Pull request merged**